FreeRTOS_DNS.c

```diff
@@ -2011,12 +1846,22 @@
         {
             BaseType_t x;
             BaseType_t xFound = pdFALSE;
-            uint32_t ulCurrentTimeSeconds = ( xTaskGetTickCount() / portTICK_PERIOD_MS ) / 1000UL;
+            TickType_t xCurrentTickCount = xTaskGetTickCount();
+            uint32_t ulCurrentTimeSeconds;
             uint32_t ulIPAddressIndex = 0;
             static BaseType_t xFreeEntry = 0;
 
             configASSERT( ( pcName != NULL ) );
 
+            if( xCurrentTickCount != 0 )
+            {
+                ulCurrentTimeSeconds = ( xCurrentTickCount / portTICK_PERIOD_MS ) / 1000UL;
+            }
+            else
+            {
+                ulCurrentTimeSeconds = 0;
+            }
+
```

Comment:
Why not just :
~~~c
    ulCurrentTimeSeconds = ( xCurrentTickCount / portTICK_PERIOD_MS ) / 1000UL;
~~~
?

When `xCurrentTickCount` is zero, the resulting `ulCurrentTimeSeconds` will also become zero.

Reply:
Oops! I made a foolish mistake here. I was mistaken - I thought that `xCurrentTickCount` was in the denominator and would cause a "division by 0" error. I will rectify that. Thanks for pointing it out.

Reply:
Fixed.